Support/ceptr/ceptr/converter.py

```diff
@@ -456,11 +459,17 @@ def mechanism_header_includes(self, fstream):
                 "#define %s_ID %d"
                 % (s, self.species_info.ordered_idx_map[species]),
             )
+            if s[-1] == "n" or s[-1] == "p" or s == "E":
+                nb_ions += 1
         cw.writer(fstream)
         cw.writer(fstream, "#define NUM_ELEMENTS %d" % (nb_elem))
         cw.writer(
             fstream, "#define NUM_SPECIES %d" % (self.species_info.n_species)
         )
+        if nb_ions > 0:
+            cw.writer(
```

Comment:
Do we just always want to define this for all mechs? If the code is using `NUM_IONS` anywhere it won't compile with any of the other mechs. So maybe all mechs should have this defined?

Reply:
yeah it's not great but it feels safer... maybe there's some other way? I can't think of one...